src/graph/executor/query/TraverseExecutor.cpp

```diff
@@ -162,8 +162,39 @@ folly::Future<Status> TraverseExecutor::handleResponse(RpcResponse&& resps) {
   }
   auto listVal = std::make_shared<Value>(std::move(list));
   auto iter = std::make_unique<GetNeighborsIter>(listVal);
-  if (currentStep_ == 1 && range_ && range_->min() == 0) {
-    result_.rows = buildZeroStepPath(iter.get());
+  if (currentStep_ == 1) {
+    if (range_ && range_->min() == 0) {
+      result_.rows = buildZeroStepPath(iter.get());
+    }
+    // match (v)-[e:Rel]-(v1:Label1)-[e1*2]->() where id(v0) in [6, 23] return v1
+    // the attributes of v1 will be obtained in the second traverse operator
+    // If the conditions are not met, the path in the previous step needs to be filtered out
+    if (!dst2PathsMap_.empty()) {
+      std::unordered_set<Value, VertexHash, VertexEqual> existVids;
+      existVids.reserve(iter->numRows());
+      auto vertices = iter->getVertices();
+      for (auto& vertex : vertices.values) {
+        if (vertex.isVertex()) {
+          existVids.emplace(vertex);
+        }
+      }
+      auto dst2PathIter = dst2PathsMap_.begin();
+      while (dst2PathIter != dst2PathsMap_.end()) {
+        if (existVids.find(dst2PathIter->first) == existVids.end()) {
+          dst2PathIter = dst2PathsMap_.erase(dst2PathIter);
```

Comment:
pay attention to the performance influence to this new added routine of iterate & erase logic, when this merged